R/pkg/R/DataFrame.R

```diff
@@ -1384,7 +1384,7 @@ setMethod("saveAsTable",
                                     "org.apache.spark.sql.parquet")
             }
             allModes <- c("append", "overwrite", "error", "ignore")
-            if (!(mode %in% allModes)) {
+            if (! (mode %in% allModes)) {
```

Comment:
Yeah lets leave the `!` out of this patch. We can add an exclusion before these two uses to get lintr to pass with no warnings.